src/crimson/osd/main.cc

```diff
@@ -217,17 +218,16 @@ int main(int argc, char* argv[])
     ("prometheus_prefix", bpo::value<string>()->default_value("osd"),
      "Prometheus metrics prefix");
 
-  auto [ceph_args, app_args] = partition_args(app, argv, argv + argc);
-  if (ceph_argparse_need_usage(ceph_args) ||
-      ceph_argparse_need_usage(app_args)) {
+  auto [ceph_args, app_args] = partition_args(argc, argv);
+  if (ceph_argparse_need_usage(app_args)) {
     usage(argv[0]);
   }
   override_seastar_opts(app_args);
   std::string cluster_name{"ceph"};
   std::string conf_file_list;
   // ceph_argparse_early_args() could _exit(), while local_conf() won't ready
   // until it's started. so do the boilerplate-settings parsing here.
-  auto init_params = ceph_argparse_early_args(ceph_args,
+  auto init_params = ceph_argparse_early_args(app_args,
```

Comment:
Why have we changed from `ceph_args` to `app_args` here?

Reply:
The meaning of app_args and ceph_args has changed slightly.  previously:

ceph_args -> everything for ceph (config_proxy + ceph_argparse_early_args)
app_args -> everything else (seastar)

but with the way we are crafting these now, it's:

ceph_args -> everything that goes to config_proxy
app_args -> everything else (ceph_argparse_early_args + seastar)

That's why we are pulling the early args from app_args now instead of ceph_args.  We could move the ceph_argparse_early_args call into partition_args and handle it there, but we need init_params later on in main so I just left it in it's current position.

Reply:
I see! Well, the naming is a little bit confusing now. How about renaming into e,g,:

* `config_proxy_args`,
* `seastar_n_early_args`?

Reply:
I actually thought the naming now sort of mimics what seastar is doing internally:  ceph for internal ceph config args and app for runtime application arguments.  I don't actually care though so will change.